Sources/SwiftSyntax/generated/SyntaxRewriter.swift

```diff
@@ -3898,79 +3909,52 @@ open class SyntaxRewriter {
 
     // newLayout is nil until the first child node is rewritten and rewritten
     // nodes are being collected.
-    var newLayout: ContiguousArray<RawSyntax?>?
+    var newLayout: UnsafeMutableBufferPointer<RawSyntax?> = .init(start: nil, count: 0)
 
-    // Rewritten children just to keep their 'SyntaxArena' alive until they are
-    // wrapped with 'Syntax'
-    var rewrittens: ContiguousArray<Syntax> = []
+    // Keep 'SyntaxArena' of rewritten nodes alive until they are wrapped
+    // with 'Syntax'
+    var rewrittens: ContiguousArray<RetainedSyntaxArena> = []
```

Comment:
What are the performance characteristics of `ContiguousArray` vs `Array` here? I would expect `Array` to be faster because it doesn’t need to move all elements if it runs out of space. And I don’t think we need the contiguous property here anymore.

Reply:
Hm, I haven't tried `Array` here. interesting.

Reply:
Array always uses contiguous storage unless it's backed by `NSArray`.
Although there's no particular reason to use `ContiguousArray`, its implementation is simpler than `Array`. So let's keep this as-is